device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/transport/IotHubTransportTest.java

```diff
@@ -29,6 +29,7 @@
 /**
  * Unit tests for IotHubTransportPacket.
  */
+@SuppressWarnings("ThrowableNotThrown")
```

Comment:
Does the warning suggest that the unit test should be throwing the declared exceptions?

Reply:
The warning is complaining because of how jmockit expects us to make a mocked method return by throwing an exception. In our tests, we do something like

```
someMockInstance.thisMethodShouldThrow();
result = new IOException();
```

The warning sees that we create an exception instance, but never throw it and complains. This is the expected way for jmockit to make mocked methods throw though, so I just suppressed these warnings